code/modules/client/client_procs.dm

```diff
@@ -204,11 +202,11 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 //This stops files larger than UPLOAD_LIMIT being sent from client to server via input(), client.Import() etc.
 /client/AllowUpload(filename, filelength)
 	if (holder)
-		if(filelength > UPLOAD_LIMIT_ADMIN)
-			to_chat(src, "<font color='red'>Error: AllowUpload(): File Upload too large. Upload Limit: [UPLOAD_LIMIT_ADMIN/1024]KiB.</font>")
+		if(filelength >  CONFIG_GET(number/upload_limit_admin))
+			to_chat(src, "<font color='red'>Error: AllowUpload(): File Upload too large. Upload Limit: [(CONFIG_GET(number/upload_limit_admin))/1024]KiB.</font>")
 			return FALSE
-	else if(filelength > UPLOAD_LIMIT)
-		to_chat(src, "<font color='red'>Error: AllowUpload(): File Upload too large. Upload Limit: [UPLOAD_LIMIT/1024]KiB.</font>")
+	else if(filelength > CONFIG_GET(number/upload_limit))
+		to_chat(src, "<font color='red'>Error: AllowUpload(): File Upload too large. Upload Limit: [(CONFIG_GET(number/upload_limit))/1024]KiB.</font>")
```

Comment:
lets cache the config values for the duration of this proc since we access it multiple times (or at least more than once). Let's also use `span_warning()` instead of manually setting the font so the string looks a bit cleaner too.

```dm
/client/AllowUpload(filename, filelength)
	var/client_max_file_size = CONFIG_GET(number/upload_limit)
	if (holder)
		var/admin_max_file_size = CONFIG_GET(number/upload_limit_admin)
		if(filelength > admin_max_file_size)
			to_chat(src, span_warning("Error: AllowUpload(): File Upload too large. Upload Limit: [admin_max_file_size/1024]KiB."))
			return FALSE
	else if(filelength > client_max_file_size)
		to_chat(src, span_warning("Error: AllowUpload(): File Upload too large. Upload Limit: [client_max_file_size/1024]KiB."))
```